src/diffusers/models/attention_processor.py

```diff
@@ -1218,11 +1226,8 @@ def __call__(
         key = key.view(batch_size, -1, attn.heads, head_dim)
         value = value.view(batch_size, -1, attn.heads, head_dim)
 
-        # Apply QK norm.
-        if attn.norm_q is not None:
-            query = attn.norm_q(query)
-        if attn.norm_k is not None:
-            key = attn.norm_k(key)
+        # Apply QK-norm if needed
+        query, key = attn.maybe_apply_qk_norm(query, key)
```

Comment:
It's just four lines of code that we're wrapping as a method. It's not doing anything complex, either. So, agree with @yiyixuxu that we should just keep it here.

Reply:
I see, I was hoping to get in a refactor based on my comment above to have some attention processors removed/reused. But we can come back to this later. Should be good to merge now? Let's do this as soon as possible to perform rebase